src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -66,11 +66,23 @@ public IStringLocalizer Create(Type resourceSource)
             var typeInfo = resourceSource.GetTypeInfo();
             var assembly = typeInfo.Assembly;
 
-            // Re-root the base name if a resources path is set
-            var baseName = string.IsNullOrEmpty(_resourcesRelativePath)
-                ? typeInfo.FullName
-                : _hostingEnvironment.ApplicationName + "." + _resourcesRelativePath
-                    + TrimPrefix(typeInfo.FullName, _hostingEnvironment.ApplicationName + ".");
+            string baseName;
+
+            if (_hostingEnvironment.ApplicationName == null)
+            {
+                var assemblyName = assembly.GetName().Name + ".";
+                //We're working in a library
+                baseName = assemblyName + _resourcesRelativePath + TrimPrefix(typeInfo.FullName, assemblyName);
```

Comment:
Don't know why I can't comment on the appropiate line, but in https://github.com/aspnet/Localization/pull/265/files#diff-ec3f46a4394df7156f745d976f053127R49 can you check if you can have a resourcesRelativePath ending in . in the options? In that case it could end up with two . on it. (It's possibly a separate issue)